WeatherApp/DAL/GeocodingRepository.cs

```diff
@@ -0,0 +1,33 @@
+﻿using DAL.Models;
+using DAL.Models.OpenWeather;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace DAL
+{
+    public class GeocodingRepository : IGeocodingRepository
+    {
+        private readonly string _apiKey;
+        private readonly HttpClient _httpClient = new HttpClient();
```

Comment:
HttpClient needs to be disposed after use. Please use "using" keyword and move HttpClient to the body of method GetCoordinatesByCityNameAsync